charts/presto/values.yaml

```diff
@@ -0,0 +1,172 @@
+# Default values for presto
+
+image:
+  repository: ahanaio/prestodb
+  pullPolicy: IfNotPresent
+  tag: ~
+
+imagePullSecrets: []
+
+service:
+  type: ClusterIP
+  port: 8080
+  nodePort: ~
+  labels: {}
+  annotations: {}
+
+ingress:
+  enabled: false
+  labels: {}
+  annotations: {}
+  ingressClassName: ~
+  tls: []
+  host: presto.local
+  path: /
+  pathType: Prefix
+
+serviceAccount:
+  create: true
+  labels: {}
+  annotations: {}
+
+# Common environment variables (templated)
+env: []
+envFrom: []
+
+# Common volumes (templated)
+volumes: []
+volumeMounts: []
+
+# Common extra and init containers (templated)
+containers: []
+initContainers: []
+
+# Constrain all presto pods to specific nodes
+nodeSelector: {}
+tolerations: []
+affinity: {}
+
+# Security context by default
+securityContext: {}
+
+# Presto home directory
+home: /opt/presto-server
+
+# Deployment mode
+# single - Minimal Presto deployment where single pod acts as Coordinator and Worker
+# cluster - Standard Presto deployment with one Coordinator and multiple Workers
+# ha-cluster - [Not supported yet] Highly available Presto deployment with multiple Resource Managers, Coordinators and Workers
+mode: cluster
+
+# Common node properties, see https://prestodb.io/docs/current/installation/deployment.html#node-properties
+node:
+  environment: development
+  dataDir: /var/presto/data
+
+# Common JVM options, see https://prestodb.io/docs/current/installation/deployment.html#jvm-config
+jvm: |-
+  -server
+  -XX:+ExitOnOutOfMemoryError
+  -Djdk.attach.allowAttachSelf=true
+
+# Common configuration properties, see https://prestodb.io/docs/current/installation/deployment.html#config-properties
+config: |-
+  query.max-memory=1GB
+  query.max-memory-per-node=512MB
```

Comment:
1. What is the reason to add `query.max-stage-count` property with the default value explicitly?

2. I didn't include `query.max-total-memory` and `query.max-total-memory-per-node` because they are not required. I know it is a good practice to set them explicitly, but the values depends on the use cases, so I would like to keep defaults because probably better values:
   - https://prestodb.io/docs/current/admin/properties.html#query-max-total-memory
   - https://prestodb.io/docs/current/admin/properties.html#query-max-total-memory-per-node

Reply:
The reason to make this configurable is that `query.max-stage-count` is being set to `100` as default and this can be not enough for more complicated queries. For example some of TPC-DS queries will fail with default setting.

Reply:
But it's configurable property, i.e. the chart user can set it in the current implementation.

Reply:
I was thinking about this settings - and even in this form it is incorrectly implemented. First of all if we make memory management configurable here the `-Xms` and `-Xmx` JAVA params and `memory.heap-headroom-per-node` Presto configuration parameter should be also set based on use-case and user-settings. We need to keep in mind that there are some checks being made on startup and:

`heap-xmx > memory.heap-headroom-per-node + query.max-total-memory-per-node`
and
`query.max-total-memory-per-node > query.max-memory-per-node`
and
`query.max-total-memory > query.max-memory`

so at least this should be set to something like 
`-Xms2g` `-Xmx2g`

We should also distinguish two basic and generic use-cases (based on `node-scheduler.include-coordinator`):

1) Test case for not including coordinator, so the value `false`
2) Test case for including coordinator, so `node-scheduler.include-coordinator=true`

Then:

1) First test case does not require overriding the coordinator config regarding `query.max-memory-per-node` and `query.max-total-memory-per-node`. It should not be included as those can be set by user to really big values (our team use values close to 1TB per node) and coordinator does not need them at all - only `query.max-memory` and `query.max-total-memory` should be set and all other can be default.

2) All settings regarding query should be configured similar to the worker nodes,

So first of all `config` should be also moved to `coordinator` and `worker` subtrees and current values should no be set as a generic setting. I would rather propose adding a `.Values` subtree named `query` like:

```yaml
query:
  enable: true # if generate config based on this or not
  heapSizePerNodeGb: 3 # must be > maxTotalMemoryPerNode + heapHeadroomPerNodeGb
  heapHeadroomPerNodeGb: 1
  maxMemoryPerNodeGb: 1 # max user memory per worker
  maxTotalMemoryPerNodeGb: 2 # default is maxMemoryPerNode * 2
  maxStageCount: 100 # max number of query stages
```
and base config generation on above. I can provide some scripts in other comment - but to sum-up this way you can create pre-deployment fail-tests and calculate some basic values like:

`query.max-memory={{ mul .Values.worker.replicas .Values.query.maxMemoryPerNodeGb }}`
`query.max-total-memory={{ mul .Values.worker.replicas .Values.query.maxTotalMemoryPerNodeGb }}`

Reply:
I see your points and I agree with you in general. 
In the current implementation it is up to a user to set a right configuration, including `-Xms` and `-Xmx` values, which I skipped intentionally for now. But definitely, resource configurations must be improved.
I'm thinking about starting from having something similar to Spark approach, like the following config
```
coordinator:
  resources:
    cpu: 4
    memory: 14G
    memoryOverhead: 0.1
```
set JVM options `-Xms14G -Xmx14G`  and Kubernetes memory requests and limits to `14G * (1 + 0.1)`.

Reply:
The above would be good enough for simple usage. 
Of course from benchmarking point of view you should also calculate the overall memory available for queries. Default approach allows query to use only 10% of node memory. But I assume this is not the case you are preparing the Charts for.